GiniVision/Classes/Core/Logger.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Logger.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 5/14/18.
+//
+
+import Foundation
+
+final class Logger {
+    
+    enum Event {
+        case error
+        case success
+        case warning
+        case custom(emoji: String)
+        
+        var value: String {
+            switch self {
+            case .error: return "❌"
+            case .success: return "✅"
+            case .warning: return "⚠️"
+            case .custom(let emoji): return emoji
+            }
+        }
+    }
+    
+    class func debug(message: String,
+                     event: Event,
+                     giniConfig: GiniConfiguration = .shared) {
+        
+        if giniConfig.debugModeOn {
+            print("[ GiniVision ](\(formattedString(from: Date()))): \(event.value) \(message)")
+        }
+    }
+    
+    class func formattedString(from date: Date) -> String {
+        let dateFormat = "dd-MM-yy hh:mm:ss"
+        let dateFormatter = DateFormatter()
```

Comment:
As far as I remember, it wasn't recommended to create Date Formatters all the time because they are "expensive" objects. You should consider having a static instance for the class.

Reply:
Didn't know about its bad performance 👍

Reply:
I removed it since it is not needed with `NSLog` (it already prints the date)

Reply:
`NSLog` might be another thing. It logs directly to `stderr`, but it's possible that a partner has a custom logging solution that writes somewhere else. And they don't want to redirect `stderr`. So it might be useful to be able to inject a logging function. But maybe I'm overthinking it.

Reply:
Ummm, could make sense although it is not very likely. I added a custom log block to the `GiniConfiguration` so they can customize it.